core/opendaq/device/include/opendaq/device_impl.h

```diff
@@ -92,6 +92,7 @@ class GenericDevice : public SignalContainerImpl<TInterface, IDeviceDomain, IDev
     ErrCode INTERFACE_FUNC removeStreamingOption(IString* protocolId) override;
     ErrCode INTERFACE_FUNC getStreamingOptions(IList** streamingOptions) override;
     ErrCode INTERFACE_FUNC setAsRoot() override;
+    ErrCode INTERFACE_FUNC setDeviceInfo(IDeviceInfo* deviceInfo) override;
```

Comment:
You should not be able to externally override the device information of a device like that. You should instead have on `IDeviceInfoInternal` a method `addServerCapability` and `removeServerCapability`.